dlt/common/schema/typing.py

```diff
@@ -91,6 +91,7 @@ class TColumnType(TypedDict, total=False):
     data_type: Optional[TDataType]
     precision: Optional[int]
     scale: Optional[int]
+    partition: Optional[bool]
```

Comment:
Please have a look at how we add partitions in bigquery with the bigquery adapter, it's very easy to do and you can copy most of the code from there probably.